### PR TITLE
chore: Storage category lint fixes

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
lint fixes in storage category

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/ddb-stack-builder.ts [hunk 1]
```diff
@@ -21,60 +24,55 @@ export class AmplifyDDBResourceStack extends cdk.Stack implements AmplifyDDBReso
...
   /**
-   *
-   * @param props :cdk.CfnOutputProps
-   * @param logicalId: : lodicalId of the Resource
+   * adds  cfn output to stack
    */
   addCfnOutput(props: cdk.CfnOutputProps, logicalId: string): void {
     try {
+      // eslint-disable-next-line no-new
       new cdk.CfnOutput(this, logicalId, props);
```
Comment (sobolk) [inline packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/ddb-stack-builder.ts:32]: out of curiosity. I noticed this pattern of few places. I.e. we `new` something out of the blue and we drop reference. Does ctor attach it somehow to `this` ?

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-transform.ts [hunk 2]
```diff
@@ -12,28 +13,31 @@ import {
...
 import * as path from 'path';
 import * as vm from 'vm2';
-import os from 'os';
 import { S3PermissionType, S3UserInputs } from '../service-walkthrough-types/s3-user-input-types';
-import { canResourceBeTransformed, S3CFNDependsOn, S3CFNPermissionType, S3InputState } from '../service-walkthroughs/s3-user-input-state';
+// eslint-disable-next-line import/no-cycle
+import {
+  canResourceBeTransformed, S3CFNDependsOn, S3CFNPermissionType, S3InputState,
+} from '../service-walkthroughs/s3-user-input-state';
+// eslint-disable-next-line import/no-cycle
```
Comment (sobolk) [inline packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-transform.ts:24]: we should log a work item to refactor those cycles. e.g. by extracting those functions to 3rd file that's then used by both modules.

**Review by sobolk — APPROVED** [review]

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-transform.ts (hunk 2 above)
Comment (akshbhu) [reply packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-transform.ts:24]: yeah , I was thinking of discussing with you for these dependency files. these are present everywhere in our code

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/ddb-stack-builder.ts (hunk 1 above)
Comment (akshbhu) [reply packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/ddb-stack-builder.ts:32]: yeah , ctor attaches this output to stack and return the value

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-builder.ts [hunk 3]
```diff
@@ -573,19 +635,19 @@ export class AmplifyS3ResourceCfnStack extends AmplifyResourceCfnStack implement
...
       const statement = new iamCdk.PolicyStatement(props);
-      //Add Statement to Policy
+      // Add Statement to Policy
       policyDocument.addStatements(statement);
     });
     return policyDocument;
   }
 
-  /***************************************************************************************************
+  /** *************************************************************************************************
```
Comment (jhockett) [inline packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-builder.ts:644]: nit: can we remove these repeated `*` lines?

**Review by jhockett — APPROVED** [review]

**Pull request merged**